### ISSUE TITLE
Failure on Ubuntu 17.04 

### ISSUE THREAD

**Issue by shamisp** [openucx/ucx#1680]
```
[ RUN      ] malloc_hook.single_thread
[     INFO ] Reduced heap free space to minimum: 4720 after allocating 10000 bytes
*** Error in `/home/pasha/ucx/build/test/gtest/gtest': double free or corruption (out): 0x0000ffff78135010 ***
Makefile:2400: recipe for target 'test' failed
```

**Comment by shamisp** [comment]
@yosefe most likely error flow is broken

**Comment by yosefe** [comment]
might be related to #1566

**Comment by shamisp** [comment]
Looks similar. I opened up GDB and I do not see anything odd. fails here:
```
258         ptr = malloc(large_alloc_size);                                                                                                                                               │
   │259         if (!RUNNING_ON_VALGRIND) {                                                                                                                                                   │
   │260             void *state = malloc_get_state();                                                                                                                                         │
   │261             malloc_set_state(state);                                                                                                                                                  │
   │262         }                                                                                                                                                                             │
  >│263         free(ptr);     
```

**Comment by shamisp** [comment]
Per @yosefe request adding backtrace:
```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
#1  0x0000ffff8fef5f5c in __GI_abort () at abort.c:89
#2  0x0000ffff8ff2d7d8 in __libc_message (do_abort=do_abort@entry=2, fmt=fmt@entry=0xffff8ffde800 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x0000ffff8ff33bf8 in malloc_printerr (action=3, str=0xffff8ffde928 "double free or corruption (out)", ptr=<optimized out>, ar_ptr=<optimized out>) at malloc.c:5048
#4  0x0000ffff8ff345f8 in _int_free (av=0xffff90006998 <main_arena>, p=0xffff8cdf6000, have_lock=0) at malloc.c:3904
#5  0x00000000004699b8 in test_thread::test (this=0xfffff84cbe98) at /home/ucx/contrib/../test/gtest/ucm/malloc_hook.cc:263
#6  0x0000000000471a84 in test_thread::thread_func (arg=<optimized out>) at /home/ucx/contrib/../test/gtest/ucm/malloc_hook.cc:85
#7  0x0000ffff902c1004 in start_thread (arg=0x471a78 <test_thread::thread_func(void*)>) at pthread_create.c:458
#8  0x0000ffff8ff8bee8 in thread_start () at ../sysdeps/unix/sysv/linux/aarch64/clone.S:87
```

**Comment by shamisp** [comment]
```
(gdb) p ptr 
$1 = (void *) 0xffff8cdf6010
(gdb) p large_alloc_size
$2 = 41943040
(gdb) 
```

**Comment by shamisp** [comment]
I just confirmed, it fails exactly in the same place on x86. So, this is ubuntu 17.04 specific issue.

**Comment by shamisp** [comment]
In order to run you need #1690

**Comment by yosefe** [comment]
should be fixed by 0a214d1371f416dac42e3b08f095bca2f4e6cc53
